tests/settings_tests/tests.py

```diff
@@ -511,3 +511,45 @@ def test_session_verification_deprecation_neither(self):
             warnings.filterwarnings('always')
             Settings('fake_settings_module')
         self.assertEqual(len(warn), 0)
+
+
+class TestAllowedHostsSettings(unittest.TestCase):
+    """
+    Make sure Allowed_Hosts setting raises an ImproperlyConfigured during
+    appropriate conditions.
+    """
+
+    def test_debug_off_unconfigured(self):
+        settings_module = ModuleType('fake_settings_module')
+        settings_module.DEBUG = False
+        settings_module.ALLOWED_HOSTS = []
+        with self.assertRaises(ImproperlyConfigured):
+            warnings.filterwarnings('always')
+            Settings(settings_module)
+
+    def test_debug_off_configured(self):
+        settings_module = ModuleType('fake_settings_module')
+        settings_module.DEBUG = False
+        settings_module.ALLOWED_HOSTS = ['localhost']
+        try:
+            Settings(settings_module)
```

Comment:
I think it's better to omit the try/except and instead say `Settings(settings_module)  # should not raise an exception`. See 071801ccff970682a799ce754431a3c3ce3d6902